src/Microsoft.AspNetCore.SignalR.Common/Internal/Formatters/BinaryMessageFormatter.cs

```diff
@@ -12,16 +11,29 @@ public static class BinaryMessageFormatter
     {
         public static void WriteMessage(ReadOnlySpan<byte> payload, Stream output)
         {
-            // TODO: Optimize for size - (e.g. use Varints)
-            var length = sizeof(long);
-            var buffer = ArrayPool<byte>.Shared.Rent(length);
-            BufferWriter.WriteBigEndian<long>(buffer, payload.Length);
-            output.Write(buffer, 0, length);
-            ArrayPool<byte>.Shared.Return(buffer);
+            var lenBuffer = new byte[5];
```

Comment:
Looks like a possible candidate for `stackalloc`? Maybe too early to think about that though

Reply:
This is a perfect case for stackalloc, soon we'll be using a new enough compiler to stackalloc `span<byte>` directly.